SequenceReplay/SequenceReplay.py

```diff
@@ -255,95 +256,103 @@ def setParameterNode(self, inputParameterNode: Optional[SequenceReplayParameterN
     def onAddButton(self) -> None:
         """Add a new sequence browser to the scene using the selected nodes as proxy nodes."""
         self._addingNodes = True
-        
+
         # Create a sequence browser node
         sequenceBrowserNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLSequenceBrowserNode")
         sequenceBrowserNode.SetPlaybackRateFps(30)
         sequencesLogic = slicer.modules.sequences.logic()
-        
+
         # Iterate through all selected nodes with their node IDs and add them to the sequence browser
         for i in range(self.ui.nodesCollapsibleButton.layout().count()):
             checkBox = self.ui.nodesCollapsibleButton.layout().itemAt(i).widget()
             if checkBox.isChecked():
                 node = slicer.mrmlScene.GetNodeByID(self.currentTransformableNodeIds[i])
-                sequenceNode = sequencesLogic.AddSynchronizedNode(None, node, sequenceBrowserNode)
-                sequenceBrowserNode.SetRecording(sequenceNode, True)
-                sequenceBrowserNode.SetPlayback(sequenceNode, True)                
-        
-        # Make sure the last added sequence browser is selected in Slicer's seqeuence browser module
+                if node:  # Ensure node is valid
+                    sequenceNode = sequencesLogic.AddSynchronizedNode(None, node, sequenceBrowserNode)
+                    sequenceBrowserNode.SetRecording(sequenceNode, True)
+                    sequenceBrowserNode.SetPlayback(sequenceNode, True)
+
+        # Make sure the last added sequence browser is selected in Slicer's sequence browser module
         slicer.modules.sequences.toolBar().setActiveBrowserNode(sequenceBrowserNode)
         self.addObserver(sequenceBrowserNode, vtk.vtkCommand.ModifiedEvent, self._updateGui)
-        
+
         self._addingNodes = False
         self._updateGui()
-    
+
     def onSequenceBrowserButtonClicked(self, sequenceBrowserNodeId: str) -> None:
-        # Make sure the clicked sequence browser is selected in Slicer's seqeuence browser module
         sequenceBrowserNode = slicer.mrmlScene.GetNodeByID(sequenceBrowserNodeId)
+        if not sequenceBrowserNode:
+            return
+
+        # Set the active browser node
         slicer.modules.sequences.toolBar().setActiveBrowserNode(sequenceBrowserNode)
-        
-        # Iterate through all sequence browser nodes. Stop replay in all except the clicked one. Start play in the clicked one.
-        for node in slicer.util.getNodesByClass("vtkMRMLSequenceBrowserNode"):
-            if node.GetID() == sequenceBrowserNodeId:
-                node.SetPlaybackActive(True)
+
+        # Toggle playback state
+        if sequenceBrowserNodeId in self.playbackState:  # If the node is currently playing
+            if self.playbackState[sequenceBrowserNodeId]:
```

Comment:
Make all string that is displayed to the user translatable. Something like this (not tested):

```
            if playback:
                buttonText = _("Play {name}").format(name=sequenceBrowserNode.GetName())
            else:
                buttonText = _("Stop {name}").format(name=sequenceBrowserNode.GetName())
            self.buttonState[sequenceBrowserNodeId].setText(buttonText)
```